src/openapi/streaming/streaming.js

```diff
@@ -445,6 +477,53 @@ function handleControlMessageResetSubscriptions(referenceIdList) {
     resetSubscriptions(subscriptionsToReset);
 }
 
+/**
+ * Handles the control message to disconnect,
+ * Notify subscriptions about connect unavailability
+ * Fire disconnect requested event
+ * @param {Array.<string>} referenceIdList
+ */
+function handleControlMessageDisconnect() {
+    log.info(
+        LOG_AREA,
+        'disconnect control message received',
+        {
+            transport: this.getActiveTransportName(),
+        },
+        {
+            persist: true,
+        },
+    );
+
+    // tell all subscriptions not to do anything
+    for (let i = 0; i < this.subscriptions.length; i++) {
+        this.subscriptions[i].onConnectionUnavailable();
+    }
+
+    this.trigger(this.EVENT_DISCONNECT_REQUESTED);
+}
+
+function handleControlMessageReconnect() {
+    log.info(
+        LOG_AREA,
+        'reconnect control message received',
+        {
+            transport: this.getActiveTransportName(),
+        },
+        {
+            persist: true,
+        },
+    );
+
+    this.isReset = true;
+
+    this.allUnsubscribePendingPromise = getAllUnsubscribePendingPromise(
+        this.subscriptions,
+    );
+
+    this.disconnect();

```

Comment:
and disconnect will always reconnect after it has finished disconnecting?

Reply:
yes.. unless we set dispose flag to true